rbc/heavydb/text_encoding_none.py

```diff
@@ -198,12 +182,58 @@ def heavydb_text_encoding_none_constructor_memcpy(context, builder, sig, args):
     return fa._getpointer()
 
 
+@extending.lower_builtin(TextEncodingNone, nb_types.UnicodeType)
+def text_encoding_none_unicode_ctor(context, builder, sig, args):
+    # Note to the future:
+    # You shall be tempted to use "context.make_helper",
+    # "cgutils.make_struct_proxy" or "builder.extract_value", but avoid it at
+    # all cost! LLVM -O2 runs SROA (Scalar Replacement Of Aggregates) and
+    # suddenly some IR nodes becomes NULL values which are deleted by DCE.
+    # Stick to plain allocas + load/store and you'll be fine.
+
+    uni_str = context.make_helper(builder, sig.args[0], value=args[0])
+
+    llty = context.get_value_type(sig.return_type.dtype)
+    st_ptr = builder.alloca(llty)
+    zero, one, two = i32(0), i32(1), i32(2)
+
+    eq = builder.icmp_signed('==', uni_str.length, uni_str.length.type(0))
+    with builder.if_else(eq, likely=False) as (then, otherwise):
+        with then:
+            bb_then = builder.basic_block
+            p1 = cgutils.get_null_value(i8p)
+            n1 = i8(1)
+        with otherwise:
+            bb_else = builder.basic_block
+            p2 = allocate_varlen_buffer(builder, uni_str.length, i64(1))
+            cgutils.raw_memcpy(builder, dst=p2, src=uni_str.data,
+                               count=uni_str.length, itemsize=i64(1))
+            # string is null terminated
+            builder.store(i8(0), builder.gep(p2, [uni_str.length]))
+            n2 = i8(0)
+
+    # {i8*, i64, i8}
+    #  ^^^
+    ptr = builder.phi(i8p)
+    ptr.add_incoming(p1, bb_then)
```

Comment:
No need to create `p1`
```suggestion
    ptr.add_incoming(NULL, bb_then)
```

Reply:
The code in the end is practically the same. For the sake of readability, let's keep `p1`.